### PR TITLE
build(webpack): correctly detect NODE_ENV

### PR DESCRIPTION
**Pull request by taymoork2**
---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "deps:generate": "node ./deps.js",
     "samples:build": "node ./tooling/index.js build --only-samples",
     "samples:serve": "NODE_ENV=development webpack serve --color",
```
Comment (lalli-flores) [inline package.json:49]: Shoudn't this get updated too?

File: webpack.config.js [hunk 2]
```diff
@@ -10,21 +10,33 @@ const {version} = require('./packages/node_modules/webex/package.json');
...
-  mode: env === 'development' ? 'development' : 'production',
+/**
+ * Webpack Config
+ *
+ * @param {object} [env]
+ * @param {string} env.NODE_ENV
+ * @param {*} [args]
+ * @returns {object}
+ */
+module.exports = (env = {NODE_ENV: process.env.NODE_ENV || 'production'}, args) => ({
```
Comment (lalli-flores) [inline webpack.config.js:21]: Wouldn't `env` come from the `args` now that you're passing it as an argument to the CLI?

**Review by lalli-flores — CHANGES_REQUESTED** [review]

**Comment by lalli-flores** [comment]
@taymoork2 make sure to also pull latest master and rebase

File: webpack.config.js (hunk 2 above)
Comment (taymoork2) [reply webpack.config.js:21]: This is more for `wdio.config.js` usage where it's imported as a config

**Comment by taymoork2** [comment]
https://gist.github.com/taymoork2/03d3b797553180d6516549edaa8fceda

File: webpack.config.js [hunk 3]
```diff
@@ -10,21 +10,30 @@ const {version} = require('./packages/node_modules/webex/package.json');
...
     './packages/node_modules/webex',
-  mode: env === 'development' ? 'development' : 'production',
+  mode: env && env.NODE_ENV === 'development' ? 'development' : 'production',
   output: {
     filename: 'webex.min.js',
     library: 'Webex',
     libraryTarget: 'umd',
     sourceMapFilename: '[file].map',
-    path: args && args.env && args.env.umd ? // samples:test fix since its called as a function
+    path: env && env.umd ? // `samples:test` script calls Webpack config as a function so `args` won't exist
```
Comment (lalli-flores) [inline webpack.config.js:30]: Seems like you got rid of `args` altogether? In that case the comment is not necessary
```suggestion
    path: env && env.umd ?
```

**Review by lalli-flores — APPROVED** [review]
Looks good, thanks for the fixes 👍🏼

**Pull request merged**